src/tool/subcommands/api_cmd.rs

```diff
@@ -971,25 +971,28 @@ async fn start_offline_server(
         start_time: chrono::Utc::now(),
         chain_store,
         beacon,
+        shutdown,
     };
     rpc_state.sync_state.write().set_stage(SyncStage::Idle);
-    start_offline_rpc(rpc_state, rpc_port).await?;
+    start_offline_rpc(rpc_state, rpc_port, shutdown_recv).await?;
 
     Ok(())
 }
 
-pub async fn start_offline_rpc<DB>(state: RPCState<DB>, rpc_port: u16) -> anyhow::Result<()>
+async fn start_offline_rpc<DB>(
+    state: RPCState<DB>,
+    rpc_port: u16,
+    mut shutdown_recv: mpsc::Receiver<()>,
+) -> anyhow::Result<()>
 where
     DB: Blockstore + Send + Sync + 'static,
 {
     info!("Starting offline RPC Server");
     let rpc_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), rpc_port);
-    let forest_version = FOREST_VERSION_STRING.as_str();
-    let (shutdown_send, mut shutdown_recv) = mpsc::channel(1);
```

Comment:
This fixes a bug, right? Was the offline RPC server correctly shutdown before?

Reply:
I don't think there was a bug here - I just needed to reorg `RPCState` to fit the new way of doing things :)